src/shogun/distributions/Gaussian.cpp

```diff
@@ -196,20 +195,29 @@ float64_t CGaussian::update_params_em(float64_t* alpha_k, int32_t len)
 	{
 	case FULL:
 	{
-		for (int32_t j = 0; j < num_dim * num_dim; j++)
-			cov_sum[j] /= alpha_k_sum;
+		linalg::scale(cov_sum, cov_sum, 1 / alpha_k_sum);
 
-		SGVector<float64_t> d0;
-		d0 = cov_sum.get_diagonal_vector();
+		SGVector<float64_t> d0(num_dim);
+#ifdef HAVE_LAPACK
+		d0.vector = SGMatrix<float64_t>::compute_eigenvectors(
```

Comment:
Please  use that method but linalg.
I think a self adjoint eigensilver is in there now
Michele will know or can help